cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,375 @@
+
+The goal of the scenario is to enable infrastructure operators to configure IRSA support for the AWS provider in Radius to deploy and manage AWS resources.
+
+## Terms and definitions
+| Term | Definition |
+|---|---|
+| OIDC | OIDC stands for OpenID Connect. It allows authentication of end-user(or workload)|
+|Role ARN| ARN associated with an AWS IAM role|
+
+
+## Objectives
+
+> Issue Reference: https://github.com/radius-project/radius/issues/7618 
+
+### Goals
+
+* Radius users can configure AWS provider and enable Radius to use IRSA for authentication and deployment of application.
+* IRSA can be configured via interactive experience.
+* IRSA can be configured with non-interactive commands.
+* Radius users can deploy and manage AWS resources without having to manage AWS access key id and secret
+
+### Non-goals
+
+* Azure Managed Identity support
+* Azure Workload Identity support
+
+The above two are addressed in [Azure Workload Identitu Support](https://github.com/radius-project/design-notes/blob/main/cli/2024-04-azure-workload-identity.md)
+
+* Ability of radified applications to be able to use IRSA for authenticating AWS operations.
+
+This facilitates a user application to be able to configure its container to use IRSA to authenticate itself for communication with AWS. While the technology is IRSA, this feature is very different from enabling Radius to use IRSA to deploy resources.
+
+### User scenarios
+
+#### User Story 1
+
+As a user I should be able to configure Radius AWS provider with IRSA through interactive experience.
+
+```
+% rad init --full  
+Enter an environment name 
+>default 
+
+% rad init --full  
+Select your cloud provider                     
+1. Azure                                  
+> 2. AWS                                     
+3.[back] 
+
+% rad init --full  
+
+Select the identity for the AWS cloud provider 
+1. IAM access key                              
+> 2. IAM Role for Service Account (IRSA)
+
+% rad init --full  
+
+Enter the IAM RoleARN for configuring IRSA: <role-arn-radius>
+```
+
+
+#### User story 2
+
+As a user I should be able to configure Radius AWS provider with IRSA using non interactive commands.
+
+1. Install the `rad` CLI.
+
+2. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.awsIRSA.enabled=true #default false
+    ```
+    If enabled, it should print a success/ failure message after updating the pod spec to mount the token. It should also print a information message to the complete of configuration by using rad credential register and rad env update.
```

Comment:
do we need the roleARN at this step? since we have to update the pod spec and change the service accounts?

Reply:
IMO we shouldn't print any information messages during rad install kubernetes, just update the docs to say what we want them to do next

Reply:
@Reshrahim and I had a discussion, where we thought we should print a message, since otherwise we have an "incomplete" setup, where user has IRSA enabled and radius expects a roleARN, and the user should use rad credentials to complete the setup. It makes sense for the behavior to be consistent across all credentials though, we can discuss this today.

Reply:
@willtsai and @Reshrahim  to decide on behavior.